src/api/error.ts

```diff
@@ -1,23 +1,25 @@
-// @flow
-
 import { log } from "../../react-native-logs.config";
 
 const logger = log.extend( "INatApiError" );
 
 export class INatApiError extends Error {
   // Object literal of the JSON body returned by the server
-  json: Object;
+  json: Record<string, unknown>;
 
   // HTTP status code of the server response
   status: number;
 
   // Additional context information
-  context: ?Object;
+  context: Record<string, unknown> | null;
 
-  constructor( json: Object, status?: number, context?: Object ) {
+  constructor(
+    json: Record<string, unknown> & { status: string | number },
+    status?: number,
+    context?: Record<string, unknown> | null
+  ) {
     super( JSON.stringify( json ) );
     this.json = json;
-    this.status = status || json.status;
+    this.status = status || +json.status;
```

Comment:
Functional change 1: This ensures the string statuses returned by the backend are always stored as number properties